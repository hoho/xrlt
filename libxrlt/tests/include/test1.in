id:0, type:0, last:0, data:
id:0, type:0, last:0, data:
id:0, type:0, last:0, data:
id:0, type:0, last:0, data:
id:0, type:0, last:0, data:
id:0, type:0, last:0, data:
id:0, type:0, last:0, data:
id:3, type:2, last:1, data:hello
id:2, type:2, last:1, data:hello
id:1, type:2, last:1, data:hello
id:4, type:2, last:1, data:hello
id:0, type:0, last:0, data:
id:5, type:2, last:1, data:hello
id:0, type:0, last:0, data:
id:6, type:2, last:1, data:hello
id:0, type:0, last:0, data:
