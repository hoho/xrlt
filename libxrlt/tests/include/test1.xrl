<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">
    <xrl:response>
        <xrl:include>
            <xrl:href>/test/href/1</xrl:href>
        </xrl:include>

        <xrl:include>
            <xrl:href>/test/href/2</xrl:href>
            <xrl:type>JSON</xrl:type>
            <xrl:success select="/*" />
        </xrl:include>

        <xrl:include>
            <xrl:href>/test/href/3</xrl:href>
            <xrl:type>TEXT</xrl:type>
        </xrl:include>

        <!--<xrl:include>
            <xrl:href select="'/test/href/2'" />
            <xrl:method>POST</xrl:method>
            <xrl:type>json</xrl:type>
            <xrl:with-param type="body" name="test">aa</xrl:with-param>
            <xrl:with-param type="body" name="test2" select="'\bb/'" />
        </xrl:include>

        <xrl:include>
            <xrl:href select="'/test/href/3'" />
            <xrl:method select="'HEAD'" />
            <xrl:type select="'xml'" />
            <xrl:with-body>bobobo</xrl:with-body>
        </xrl:include>

        <xrl:include>
            <xrl:href select="'/test/href/4'" />
            <xrl:method>put</xrl:method>
            <xrl:type select="'text'" />
            <xrl:with-body test="false()">bobobo</xrl:with-body>
        </xrl:include>

        <xrl:include>
            <xrl:href select="'/test/href/5'" />
            <xrl:method select="'HEAD'" />
            <xrl:with-body>
                <xrl:test select="true()" />
                <xrl:value>bababa</xrl:value>
            </xrl:with-body>
        </xrl:include>

        <xrl:include>
            <xrl:href select="'/test/href/6'" />

            <xrl:with-header name="he1" select="'va1'" />

            <xrl:with-header name="he2" test="false()" select="'va2'" />

            <xrl:with-header>
                <xrl:name>he3</xrl:name>
                <xrl:value>va3</xrl:value>
            </xrl:with-header>

            <xrl:with-header>
                <xrl:test select="false()" />
                <xrl:name>he4</xrl:name>
                <xrl:value>va4</xrl:value>
            </xrl:with-header>

            <xrl:with-param name="pa1" select="'pava1'" />

            <xrl:with-param name="pa2" test="false()" select="'pava2'" />

            <xrl:with-param>
                <xrl:name>pa3</xrl:name>
                <xrl:value>p,a;v&amp;a=3</xrl:value>
            </xrl:with-param>

            <xrl:with-param>
                <xrl:test></xrl:test>
                <xrl:name>pa4</xrl:name>
                <xrl:value>pava4</xrl:value>
            </xrl:with-param>

            <xrl:with-param>
                <xrl:test>true</xrl:test>
                <xrl:name>pa5</xrl:name>
                <xrl:value>pava5</xrl:value>
            </xrl:with-param>

            <xrl:with-body>bobobo</xrl:with-body>
        </xrl:include>-->

    </xrl:response>
</xrl:requestsheet>
