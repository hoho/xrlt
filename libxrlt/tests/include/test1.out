XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: XML
sr url: /test/href/1
sr query: (null)
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 2
sr method: HEAD
sr type: XML
sr url: /test/href/3
sr query: (null)
sr body: bobobo
XRLT_STATUS_SUBREQUEST
sr id: 3
sr method: PUT
sr type: TEXT
sr url: /test/href/4
sr query: (null)
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 4
sr method: HEAD
sr type: XML
sr url: /test/href/5
sr query: (null)
sr body: bababa
XRLT_STATUS_SUBREQUEST
sr id: 5
sr method: POST
sr type: JSON
sr url: /test/href/2
sr query: (null)
sr body: test=aa&test2=%5Cbb%2F
XRLT_STATUS_SUBREQUEST
sr id: 6
sr method: GET
sr type: XML
sr header: he1: va1
sr header: he3: va3
sr url: /test/href/6
sr query: pa1=pava1&pa3=p%2Ca%3Bv%26a%3D3&pa5=pava5
sr body: bobobo
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
